src/co2.js

```diff
@@ -83,7 +92,7 @@ class CO2 {
    *
    * @param {number} bytes
    * @param {boolean} green
-   * @return {number} the amount of CO2 in grammes
+   * @return {number|CO2EstimateComponents} the amount of CO2 in grammes or its separate components
```

Comment:
i think there are cases where `perVisit` has a different return type than `perByte`:

```js
const co2Emission = new co2({
  results: "segment",
});

const bytesSent = 1000 * 1000 * 1000;
const greenHost = false;

console.log(co2Emission.perVisit(bytesSent, greenHost));
```

the result i'm seeing looks like this:

```json
{
    "consumerDeviceCO2 - first": 154.825749,
    "consumerDeviceCO2 - subsequent": 1.0321716600000002,
    "networkCO2 - first": 41.68385550000001,
    "networkCO2 - subsequent": 0.27789237000000006,
    "productionCO2 - first": 56.570946750000004,
    "productionCO2 - subsequent": 0.3771396450000001,
    "dataCenterCO2 - first": 44.66127375,
    "dataCenterCO2 - subsequent": 0.297741825,
    "total": 299.7267705
}
```

Reply:
Good spot, I'll have a look at that - might need a unique type for `perVisit`.  
Is there possibly also a bug there? Adding up all of the first values only gives ~297.74, so I assume the total is including the subsequent visit information too. In terms of what you might want to use this for it would make more sense to have `"total - first"` and `"total - subsequent"` properties. Or use sub-objects with first and subsequent values for all of them.

Reply:
NVM, I see that both parts of the component are reduced by a percentage so the total is correct. Having separated total components would be a feature request/matter of taste.